src/triage/component/catwalk/evaluation.py

```diff
@@ -106,6 +111,70 @@ def __init__(
     def sessionmaker(self):
         return sessionmaker(bind=self.db_engine)
 
+    def _subset_labels_and_predictions(
```

Comment:
Looking at these two private methods, I think they probably belong as their own functions or classes. A PredictionSubsetter class or something, or even just a function (`generate_binary_at_x` in this file is an example of a regular function that is scoped similarly to `subset_labels_and_predictions`). Why do I say this? First, looking at the test it was a code smell that you were unit testing a private method. Testing a private method can either mean over-testing or too much functionality in one class (I'd argue this is the latter).

To support this hypothesis, neither of these private methods really use anything from the ModelEvaluator class. There's nothing really tying them to here, other than the fact that they are called here.

Reply:
I actually had the same thoughts as I was writing them but decided to leave them in. They really only use the db_engine, though, which can easily be passed.